node/impl/full/state.go

```diff
@@ -449,6 +450,21 @@ func (m *StateModule) StateLookupID(ctx context.Context, addr address.Address, t
 	return m.StateManager.LookupID(ctx, addr, ts)
 }
 
+func (m *StateModule) StateLookupRobustAddress(ctx context.Context, addr address.Address, tsk types.TipSetKey) (address.Address, error) {
+	ts, err := m.Chain.GetTipSetFromKey(ctx, tsk)
+	if err != nil {
+		return address.Undef, xerrors.Errorf("loading tipset %s: %w", tsk, err)
+	}
+	if ts.Height() > policy.ChainFinality {
```

Comment:
So, thinking about this some more, we aren't actually using a finality lookback when resolving `Account`s in `StateAccountKey`. While I think we probably should be, this method should just match behaviour there, so I'd actually drop the finality lookback here (in contrast to what I said we should do in the issue).